### PR TITLE
Adds telegraf image in the beryllium stack. The config is also generated using the env variable

### PR DESCRIPTION
**Pull request by eoliveros**
this only grabs the host cpu, host mem, host disk. not including blueprint or anything like that.

### CODE REVIEW

File: telegraf/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FROM telegraf:latest
```
Comment (djpnewton) [inline telegraf/Dockerfile:1]: lets lock this to a specific version number as we have a very security conscious app

File: docker-compose.yml [hunk 2]
```diff
@@ -153,6 +153,22 @@ services:
...
+  telegraf:
+    image: telegraf
+    container_name: telegraf
+    environment:
+      TELEGRAF_NAME: ${TELEGRAF_NAME}
+      TELEGRAF_INFLUXDB_DB: ${TELEGRAF_INFLUXDB_DB:?TELEGRAF_INFLUXDB_DB unset}
+      TELEGRAF_INFLUXDB_URL: ${TELEGRAF_INFLUXDB_URL:?TELEGRAF_INFLUXDB_URL unset}
+      TELEGRAF_INFLUXDB_USER: ${TELEGRAF_INFLUXDB_USER:?TELEGRAF_INFLUXDB_USER unset}
+      TELEGRAF_INFLUXDB_PASS: ${TELEGRAF_INFLUXDB_PASS:? TELEGRAF_INFLUXDB_PASS unset}
+    volumes:
```
Comment (djpnewton) [inline docker-compose.yml:165]: This looks like it will not be monitoring the host because the system directories arent mounted (https://hub.docker.com/_/telegraf?tab=description "Monitoring the Docker Engine Host")

**Review by djpnewton — COMMENTED** [review]
I dont see any reference in telegraf to http://be_app ?

File: telegraf/Dockerfile (hunk 1 above)
Comment (eoliveros) [reply telegraf/Dockerfile:1]: will change it to be the same as the others being used on servers deployed previously.

File: docker-compose.yml (hunk 2 above)
Comment (eoliveros) [reply docker-compose.yml:165]: looks like i need to add
```
-H "HOST": "localhost"
```
to my curl query to get it to work.

File: telegraf/Dockerfile (hunk 1 above)
Comment (eoliveros) [reply telegraf/Dockerfile]: changed it to match a specific version rather than :latest

File: docker-compose.yml (hunk 2 above)
Comment (djpnewton) [reply docker-compose.yml:174]: I still dont see the host system directories mounted

File: telegraf/entrypoint-shim.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+set -e
+
+jinja2 /telegraf.conf.j2 > "/etc/telegraf/telegraf.conf"
+
+if [ "${1:0:1}" = '-' ]; then
```
Comment (djpnewton) [inline telegraf/entrypoint-shim.sh:6]: this file needs some comments, it is not clear what is happening

File: src/monitoring_endpoint.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    ln_info['total_receivable'] = _msat_to_sat(total_receivable)
+    ln_info['overall_total_sats'] = _msat_to_sat(overall_total)
+    balance = LnRpc().list_funds()
+    ln_info['sats_channels'] = balance['sats_channels']
+    ln_info['sats_onchain'] = balance['sats_onchain']
+    dasset_balances = account_balances()
+    for balance in dasset_balances or []:
+        ln_info[f'{balance.symbol}_available'] = format(balance.available, f'.{balance.decimals}f')
+        ln_info[f'{balance.symbol}_total'] = format(balance.total, f'.{balance.decimals}f')
+    return ln_info
```
Comment (djpnewton) [inline src/monitoring_endpoint.py:64]:
```suggestion

    ln_funds = LnRpc().list_funds()
    ln_info = LnRpc().get_info()
    dasset_balances = account_balances()

    info = {}
    info['total_spendable'] = _msat_to_sat(total_spendable)
    info['total_receivable'] = _msat_to_sat(total_receivable)
    info['overall_total_sats'] = _msat_to_sat(overall_total)
    info['sats_channels'] = ln_funds['sats_channels']
    info['sats_onchain'] = ln_funds['sats_onchain']
    info['blockheight'] = ln_info['blockheight']
    for balance in dasset_balances or []:
        info[f'{balance.symbol}_available'] = format(balance.available, f'.{balance.decimals}f')
        info[f'{balance.symbol}_total'] = format(balance.total, f'.{balance.decimals}f')
    return info
```

**Review by djpnewton — CHANGES_REQUESTED** [review]

File: telegraf/entrypoint-shim.sh (hunk 3 above)
Comment (eoliveros) [reply telegraf/entrypoint-shim.sh:6]: as for the host variable... need to look above the TELEGRAF variables.

**Comment by eoliveros** [comment]
adding block for /monitoring in nginx:
```
  location /monitoring {
    deny all;
  }
```
Testing.

File: web/nginx/app.conf.template [hunk 5]
```diff
@@ -63,4 +63,8 @@ server {
     proxy_cache_bypass $http_upgrade;
     proxy_redirect off;
   }
+
+  location /monitoring {
```
Comment (djpnewton) [inline web/nginx/app.conf.template:67]: lets move this above beside the /spark and /alloy location blocks. it will keep it consistent and just in case there is any precedence rules

File: docker-compose.yml [hunk 6]
```diff
@@ -144,6 +144,7 @@ services:
 
   mail:
     container_name: mail
+    restart: unless-stopped
```
Comment (djpnewton) [inline docker-compose.yml:147]: this is unrelated to this PR

File: docker-compose.yml [hunk 7]
```diff
@@ -153,6 +154,31 @@ services:
...
+    environment:
+      HOST_ETC: /hostfs/etc
+      HOST_PROC: /hostfs/proc
+      HOST_SYS: /hostfs/sys
+      HOST_VAR: /hostfs/var
+      HOST_RUN: /hostfs/run
+      HOST_MOUNT_PREFIX: /hostfs
+      SERVER_NAME: ${SERVER_NAME:?SERVER_NAME unset}
+      TELEGRAF_NAME: ${TELEGRAF_NAME}
+      TELEGRAF_INFLUXDB_DB: ${TELEGRAF_INFLUXDB_DB:?TELEGRAF_INFLUXDB_DB unset}
```
Comment (djpnewton) [inline docker-compose.yml:170]: is there any way we can reduce the number of environment varaibles we need to add?

it would be nice if we could just do: `TELEGRAF=https://<user>:<pass><url>/<db_name>` or something

**Review by djpnewton — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 7 above)
Comment (eoliveros) [reply docker-compose.yml:170]: not sure. could investigate but that could delay this further when we have other things to consider ie. kapacitor stuff.

File: web/nginx/app.conf.template (hunk 5 above)
Comment (eoliveros) [reply web/nginx/app.conf.template]: moved between alloy and spark

File: docker-compose.yml (hunk 6 above)
Comment (eoliveros) [reply docker-compose.yml]: removed and will be placed on another pr to itself at a later time.

**Comment by eoliveros** [comment]
attempted to fix pylint and it was fixed on PR #74 so reset the repo to the previous commit.

**Review by djpnewton — DISMISSED** [review]

**Comment by djpnewton** [comment]
just need to squash the two commits

File: docker-compose.yml [hunk 8]
```diff
@@ -153,6 +153,31 @@ services:
...
+      TELEGRAF_INFLUXDB_URL: ${TELEGRAF_INFLUXDB_URL:?TELEGRAF_INFLUXDB_URL unset}
+      TELEGRAF_INFLUXDB_USER: ${TELEGRAF_INFLUXDB_USER:?TELEGRAF_INFLUXDB_USER unset}
+      TELEGRAF_INFLUXDB_PASS: ${TELEGRAF_INFLUXDB_PASS:? TELEGRAF_INFLUXDB_PASS unset}
+    volumes:
+      - '/:/hostfs:ro'
+      - './telegraf/telegraf_data:/etc/telegraf'
+      - './telegraf/telegraf_data/telegraf.d:/etc/telegraf/telegraf.d'
+      - './telegraf/telegraf_data/telegraf_scripts:/opt/telegraf_scripts'
```
Comment (djpnewton) [inline docker-compose.yml:177]: what are these used for?

Comment (eoliveros) [reply docker-compose.yml:177]: to define where the telegraf.conf,location of the external scripts and the external configs for the scripts

**Comment by djpnewton** [comment]
added a commit:

```
-reduce number of telegraf environment variables
-document telegraf Dockerfile
-dont replace upstream telegraf image entrypoint.sh
-rename monitoring_endpoint.py to monitor_endpoint.py and remove unused imports
```

**Comment by djpnewton** [comment]
you have force pushed a change to my commit,

- you should not force push when two people are working on a branch, unless all the work is finished and everyone is happy with it (then someone could rebase, squash etc)
- the change you have added logically applies to your previous commit (https://github.com/zap-me/beryllium/pull/70/commits/f524cff2ba5922807f62c19efce903d028d91039)
- you should have just added another commit to the end of the branch and then once we decide we are happy with the changes we can rearrange that commit and squash it into your original commit

**Comment by djpnewton** [comment]
I have force pushed my version again (to restore the history from yesterday). now you can add your new change as a new commit and we can review and decide if we are happy with the change set.

**Comment by eoliveros** [comment]
hmmm.... i git reset to 5cc6ca4 so whatever changes i did afterwards forgotten. will make the change and do a "git push" afterwards.

**Review by djpnewton — APPROVED** [review]
Ok I think this is ready to merge

@eoliveros all finished?

**Comment by eoliveros** [comment]
> Ok I think this is ready to merge
> 
> @eoliveros all finished?

yes, that should be it.
